lib/ui/pages/home.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_speed_dial/flutter_speed_dial.dart';
+import 'package:mobile_test/ui/components/standard_page.dart';
+
+import 'package:mobile_test/ui/pages/barcode.dart';
+import 'package:mobile_test/ui/pages/scanner.dart';
+
+/// The Home class represents the Home landing page. This page contains only a
+/// StandardPage Widget with no content and a Floating Action Button that the
+/// user uses to navigate to the Scan and QR Code pages.
+class Home extends StatelessWidget {
+  const Home({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return StandardPage(
+      title: 'Home',
+      body: Container(),
+      floatingActionButton: SpeedDial(
```

Comment:
**question:** what would your approach be to building this widget yourself? I'm only looking for high-level architecture here, no need to get granular.

Reply:
I assume you're referring to the SpeedDial FAB?

Usually I use a Stack to layer several Widgets one on top of another.

For instance, I use a stack to make a messaging system sit on top of the rest of the application. The large part of the Widget is a tap-through widget except for the messages which are tappable.

The rest of this Widget would be a combination of Rows / Stacks and animations that show/hide and move the widgets around.